src/mtev_cluster.c

```diff
@@ -65,13 +82,16 @@ struct mtev_cluster_t {
   mtev_cht_t **cht;
   mtev_net_heartbeat_ctx *hbctx;
   mtev_cluster_node_t *oldest_node;
+  struct jl_array_list *hb_payloads;
```

Comment:
What do you mean by `I see no reason for the arrays to every shrink`? I don't see where `jl_array_list` shrinks. 
So if this `mtev_arraylist` anything different to what you typically expect from an arraylist, I'd suggest to rename this to e.g. `mtev_json_list` and maybe create a new `mtev_arraylist` which does exactly what you suggested. But before I do that I'd really like to understand what's so special about the current implementation. The only thing I'd change is not storing an array of pointers but an array of structs to have a better cash locality.